### ISSUE TITLE
ARAP parameterizer fails on very simple scenarios

### ISSUE THREAD

**Issue by jasjuang** [CGAL/cgal#4290]
## Issue Details

Please see below minimal code to reproduce the problem
```
  using Kernel            = CGAL::Exact_predicates_inexact_constructions_kernel;
  using Point_2           = Kernel::Point_2;
  using SurfaceMesh       = CGAL::Surface_mesh<Kernel::Point_3>;
  using vertex_descriptor = boost::graph_traits<SurfaceMesh>::vertex_descriptor;
  using halfedge_descriptor =
      boost::graph_traits<SurfaceMesh>::halfedge_descriptor;
  namespace SMP = CGAL::Surface_mesh_parameterization;

  SurfaceMesh sm;

  std::ifstream input("mesh.off");
  input >> sm;
  input.close();

  halfedge_descriptor bhd =
      CGAL::Polygon_mesh_processing::longest_border(sm).first;

  SurfaceMesh::Property_map<vertex_descriptor, Point_2> uv_map =
      sm.add_property_map<vertex_descriptor, Point_2>("v:uv").first;

  using Parameterizer = SMP::ARAP_parameterizer_3<SurfaceMesh>;

  SMP::Error_code err = SMP::parameterize(sm, Parameterizer(), bhd, uv_map);

  if (err != SMP::OK)
  {
    std::cerr << "Error: " << SMP::get_error_message(err) << std::endl;
  }
```

mesh.off is a simple mesh with 4 triangles and non-complex shape

```
OFF
6 4 0
25.3639 -7.81882 164.672
-62.6845 -15.2123 137.962
-48.3032 53.8243 180.59
-165.968 9.19233 97.0192
-183.449 -37.6654 14.1766
-102.155 9.10969 141.671
3 3 1 5
3 5 1 2
3 0 2 1
3 4 1 3
```

When I run the minimal code, the below error message shows up

```
Could not solve linear system
Error: Parameterization failed: no one-to-one mapping
```

Is this expected?

## Environment

* Operating system: Ubuntu 18.04
* Compiler: GCC 7
* Release or debug mode: Release
* CGAL version: Latest
* Boost version: 1.68

**Comment by sloriot** [comment]
Could you please try #4276?

**Comment by MaelRL** [comment]
Solved by #4276 (uses more powerful solvers).

**Comment by jasjuang** [comment]
Thanks. I can confirm that #4276 fixes this issue. Looking forward to the merge.

**Comment by lrineau** [comment]
Fixed by merge of PR #4276.
